mlflow/models/evaluation/default_evaluator.py

```diff
@@ -258,6 +259,38 @@ def _do_plot(**kwargs):
 }
 
 
+def _evaluate_custom_metric(custom_metric, eval_df, builtin_metrics):
+    result = custom_metric(eval_df, builtin_metrics)
+    if result is None:
+        _logger.warning(f"Custom metric function '{custom_metric.__name__}' returned None.")
+        return None, None
+
+    def __validate_metrics_dict(metrics):
+        return isinstance(metrics, dict) and all(
+            isinstance(metric_name, str) and isinstance(metric_val, (int, float, np.number))
+            for metric_name, metric_val in metrics.items()
+        )
+
+    def __validate_artifacts_dict(artifacts):
+        return isinstance(artifacts, dict) and all(
+            isinstance(artifacts_name, str) for artifacts_name in artifacts.keys()
+        )
+
+    if isinstance(result, dict) and __validate_metrics_dict(result):
+        return result, None
+
+    if (
+        isinstance(result, tuple)
+        and __validate_metrics_dict(result[0])
+        and __validate_artifacts_dict(result[1])
+    ):
+        return result
+
+    raise MlflowException(
+        f"Custom metric '{custom_metric.__name__}' did not return in an expected format."
```

Comment:
Can we include `error_code=INVALID_PARAMETER_VALUE` here, indicating that the supplied `custom_metric` is invalid?